Makefile.COMMON

```diff
@@ -49,19 +49,22 @@ ifeq ($(GOOS),darwin)
 endif
 
 GO_VERSION ?= 1.4.2
-
-ifeq ($(shell type go >/dev/null && go version | sed 's/.*go\([0-9.]*\).*/\1/'), $(GO_VERSION))
-	GOROOT := $(shell go env GOROOT)
-else
-	GOROOT := $(CURDIR)/.build/go$(GO_VERSION)
-endif
-
 GOURL      ?= https://golang.org/dl
 GOPKG      ?= go$(GO_VERSION).$(GOOS)-$(GOARCH)$(RELEASE_SUFFIX).tar.gz
 GOPATH     := $(CURDIR)/.build/gopath
-GOCC       ?= $(GOROOT)/bin/go
-GO         ?= GOROOT=$(GOROOT) GOPATH=$(GOPATH) $(GOCC)
-GOFMT      ?= $(GOROOT)/bin/gofmt
+
+# Check for the correct version of go in the path. If we find it, use it.
+# Otherwise, prepare to build go locally.
+ifeq ($(shell type go &>/dev/null && go version | sed 's/.*go\([0-9.]*\).*/\1/'), $(GO_VERSION))
+	GOCC   ?= $(shell type -P "go")
```

Comment:
Still not sure about the details of who gets into which branch of this (I can't make it use the system go at all on a new, clean ubuntu 14.04), but `type -P` is indeed not portable enough.

As far as I can tell, `make` doesn't actually call the shell in all cases here; in at least one case (Debian Wheezy) it failed for me because it tried to call a `type` binary that doesn't exist separately.

I would suggest using `which` here – as far as I can remember this works on almost every OS.